### DISCUSSION TITLE
Multitenant with schema for each tenant

### DISCUSSION THREAD

**Discussion by wimdevriendt** [JonPSmith/AuthPermissions.AspNetCore#107] [General]
Hi, is there a possibility to use the library with a sql server database which is multitenant where all tenants have a seperate schema. So the datakey is the schemaname
Can the entityframework handle this ?

**Comment by JonPSmith** [comment]
hi @wimdevriendt,

The AuthP library doesn't support "tenant per schema" approach of building multi-tenant applications. That's because AuthP uses EF Core which uses [Global Query Filters](https://learn.microsoft.com/en-us/ef/core/querying/filters). Also the AuthP library supports other multi-tenant approaches that the "tenant per schema" can't support, such as [hierarchical](https://github.com/JonPSmith/AuthPermissions.AspNetCore/wiki/Multi-tenant-explained#hierarchicaltenant-multi-tenant) and [sharding](https://github.com/JonPSmith/AuthPermissions.AspNetCore/wiki/Sharding-explained).
